src/coreclr/src/vm/threadsuspend.cpp

```diff
@@ -3896,138 +3799,19 @@ HRESULT ThreadSuspend::SuspendRuntime(ThreadSuspend::SUSPEND_REASON reason)
-            {
-                if (thread == pCurThread)
-                    continue;
-
-                if ((thread->m_State & Thread::TS_GCSuspendPending) == 0)
-                    continue;
-
-                if (!thread->m_fPreemptiveGCDisabled)
-                    continue;
-
-#if defined(FEATURE_HIJACK) && !defined(TARGET_UNIX)
-            RetrySuspension2:
-#endif
-                // We can not allocate memory after we suspend a thread.
-                // Otherwise, we may deadlock the process when CLR is hosted.
-                ThreadStore::AllocateOSContext();
-
-#ifdef TIME_SUSPEND
-                DWORD startSuspend = g_SuspendStatistics.GetTime();
-#endif
-
-                Thread::SuspendThreadResult str = thread->SuspendThread();
-
-#ifdef TIME_SUSPEND
-                g_SuspendStatistics.osSuspend.Accumulate(
-                    SuspendStatistics::GetElapsed(startSuspend,
-                                                  g_SuspendStatistics.GetTime()));
-
-                if (str == Thread::STR_Success)
-                    g_SuspendStatistics.cntOSSuspendResume++;
-                else
-                    g_SuspendStatistics.cntFailedSuspends++;
-#endif
-
-#if defined(FEATURE_HIJACK) && !defined(TARGET_UNIX)
-                // Only check HandledJITCase if we actually suspended the thread, and
-                // the thread is in cooperative mode.
-                // See comment at the previous invocation of HandledJITCase - it does
-                // more than you think!
-                if (str == Thread::STR_Success && thread->m_fPreemptiveGCDisabled)
-                {
-                    Thread::WorkingOnThreadContextHolder workingOnThreadContext(thread);
-                    if (workingOnThreadContext.Acquired() && thread->HandledJITCase())
-                    {
-                        // Redirect thread so we can capture a good thread context
-                        // (GetThreadContext is not sufficient, due to an OS bug).
-                        if (!thread->CheckForAndDoRedirectForGC())
-                        {
-#ifdef TIME_SUSPEND
-                            g_SuspendStatistics.cntFailedRedirections++;
-#endif
-                            STRESS_LOG1(LF_SYNC, LL_INFO1000, "Failed to CheckForAndDoRedirectForGC(). Retry suspension 2 for thread %p\n", thread);
-                            thread->ResumeThread();
-                            goto RetrySuspension2;
-                        }
-#ifdef TIME_SUSPEND
-                        else
-                            g_SuspendStatistics.cntRedirections++;
-#endif
-                    }
-                }
-#endif // FEATURE_HIJACK && !TARGET_UNIX
-
-                if (str == Thread::STR_Success)
-                    thread->ResumeThread();
-            }
-#endif // DISABLE_THREADSUSPEND
-        }
-        else
-        if (res == WAIT_OBJECT_0)
+       }
+        else if (res == WAIT_OBJECT_0)
```

Comment:
if we timed out, we want to redo hijacks, no progress was made in 10msec. we set observeOnly = false, so that we redo hijacks, and reset the event here.